### ISSUE TITLE
After the orleans transaction is committed, some tasks need to be processed. How should this be done?

### ISSUE THREAD

**Issue by HermesNew** [dotnet/orleans#7883]
After the orleans transaction committed,We will do some other task.
I have no way of knowing the result of the execution of the transaction. It is also impossible to judge whether the transaction was successful or not.
How should this be handled?Can you provide some suggestions?

**Comment by ElanHasson** [comment]
Hi @HermesNew,

This new functionality may be what you're looking for: https://github.com/dotnet/orleans/pull/7805

Keep in mind it is available only in 4.0.

**Comment by HermesNew** [comment]
@ElanHasson In fact, orleans transaction already has a TransactionAgent, but there is no callback method after the transaction is completed.
I think just implement ITransactionAgent and replace the original implementation.

**Comment by HermesNew** [comment]
Of course, the best solution is to open a action for transaction completion.

**Comment by HermesNew** [comment]
> #7805 
This solution is great implementation, just don't know when it will be integrated into the version.

**Comment by ElanHasson** [comment]
I don't think it'll be in 3.x :(
